src/jit/emitxarch.cpp

```diff
@@ -10793,6 +10863,11 @@ size_t emitter::emitOutputInstr(insGroup* ig, instrDesc* id, BYTE** dp)
             assert(code & 0x00FF0000);
 
 #ifdef FEATURE_AVX_SUPPORT
+            if (TakesRexWPrefix(ins, size))
+            {
+                code = AddRexWPrefix(ins, code);
+            }
+
             if (TakesVexPrefix(ins))
```

Comment:
@BruceForstall - for AVX instructions that encode REX-equivalent bits (e.g. the W bit), we handle them as for the REX encoding. In fact, for instructions that have both AVX and SSE encodings, we simply generate the AVX encoding from the same encoding bits when AVX encodings are enabled. So, this is the universal way to set 'W' bits, whether they wind up in an AVX or VEX prefix.
I find this approach preferable to adding a separate method. It is what I did for the vgather variants when I was experimenting with that, which also require both the W and L bits to be set appropriately.